src/cov.jl

```diff
@@ -153,3 +153,37 @@ function cor2cov!(C::AbstractMatrix, s::AbstractArray)
     end
     return C
 end
+
+"""
+    CovarianceEstimator
+
+Abstract type for covariance estimators.
+"""
+abstract type CovarianceEstimator end
+
+"""
+    cov(ce::CovarianceEstimator, x::AbstractVector)
+
+Compute the sample variance of the vector `x` using estimator `ce`.
+"""
+cov(ce::CovarianceEstimator, x::AbstractVector) =
+    error("cov is not defined for $(typeof(ce)) and $(typeof(x))")
```

Comment:
How about providing a default estimator which would just call the standard `cov` function?

It would also make sense to add weights as an optional argument (in brackets), for consistency with `cov`.

Reply:
Sure, I've got a default estimator here: https://github.com/mateuszbaran/CovarianceEstimation.jl/blob/master/src/simplecov.jl , so I can just move it to StatsBase.

Weights are a bit more problematic. I don't know how to make a generic implementation that works for all estimators, so most of them would not support it anyway. In CovarianceEstimation.jl I've put it in as a part of estimator but I can change it if you think it's a good idea. But estimator objects are not (and can't) be always data-independent anyway.

Reply:
Regarding the `Simple` estimator, that's more code that I expected. Why can't it just call `cov`? Also the docs probably don't need to be repeated and can just point at the default method.

It's fine if only some estimators support weights. We just need to define a standard interface here. I'd rather keep weights out of the estimator object, passing them as an argument: they depend on the data, so you generally can't reuse an estimator for several operations if it stores weights.

Reply:
OK, makes sense. I'll trim the code and documentation.

I'll extract weights to an additional positional argument of `cov`. BTW, would you prefer to have `corrected` as a keyword argument? It somewhat depends on the data as well.

Reply:
Having `corrected` in the estimator sounds fine, I'd say it depends more on the setting than on the actual data. It could almost be considered as a separate estimator.